### PR TITLE
hologram: 8d86e3f -> d20d1c3

### PR DESCRIPTION
**Pull request by nand0p**
- Updates dependencies
- Adds configuration module
- Tested on Nixos Unstable

### CODE REVIEW

File: nixos/modules/services/security/hologram-agent.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        type        = types.str;
+        default     = "80";
+        description = "Port for metadata service to listen on.";
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+    boot.kernelModules = [ "dummy" ];
```
Comment (copumpkin) [inline nixos/modules/services/security/hologram-agent.nix]: Hmm? What's up with this

File: nixos/modules/services/security/hologram-agent.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    systemd.services.hologram-agent = {
+      description = "Provide EC2 instance credentials to machines outside of EC2";
+      after       = [ "network.target" ];
+      wantedBy    = [ "multi-user.target" ];
+      preStart = ''
+        /run/current-system/sw/bin/rm -fv /var/run/hologram.sock
+        /run/current-system/sw/bin/ip addr add 169.254.169.254/32 dev dummy0 || true
```
Comment (copumpkin) [inline nixos/modules/services/security/hologram-agent.nix]: This feels like it should be managed by config rather than the service start. #7227

Comment (nand0p) [inline nixos/modules/services/security/hologram-agent.nix]: I can verify networking.interfaces does not take when declared in config:

>    networking.interfaces.dummy0 = {
>      ipAddress = "169.254.169.254";
>      prefixLength = 32;
>    };

I also tried unused physical device and secondary ip's with ":" syntax.  in all cases, i see the systemd proc exec'd, but [virtual] device is not IP'ed.

> building path(s) ‘/nix/store/ihlgckshi7ig1spdsyccl89b7yiy1an8-unit-network-addresses-enp2s0-1.service’
> building path(s) ‘/nix/store/ivjb7b3brz99aphn90xnkq1ac8rcv80f-unit-network-link-enp2s0-1.service’

so this fails:

`[ERROR  ] 2017-02-02T10:26:39-05:00 (/tmp/nix-build-hologram-20170130-d20d1c3.drv-0/go/src/github.com/AdRoll/hologram/cmd/hologram-agent/main.go:77) Could not start up the metadata interface: listen tcp 169.254.169.254:80: bind: cannot assign requested address`

when IP'ing in `preStart`, it works.

**Comment by nand0p** [comment]
@copumpkin the problem appears to be, when switching configurations, the virtual device systemd unit files were being written out but not exec'd. By running `systemctl` in `preStart` the `dummy0` device is up and IP'd before hologram-agent runs.

**Comment by nand0p** [comment]
@copumpkin ready_for_review

**Comment by copumpkin** [comment]
Looks great, thanks!

**Pull request merged**